testSrc/unit/io/flutter/logging/FlutterErrorHelperTest.java

```diff
@@ -34,27 +34,84 @@ public void testGetAnalyticsId_scrubbing() {
     assertEquals(
       "failed-assertion",
       FlutterErrorHelper.getAnalyticsId(
-        "package:flutter/src/widgets/framework.dart':\nfailed assertion: line 123 pos 123: 'owner._debugcurrentbuildtarget == this': is not true"));
+        "'package:flutter/src/widgets/framework.dart':\n" +
+        "failed assertion: line 123 pos 123: 'owner._debugcurrentbuildtarget == this': is not true"));
     assertEquals(
       "failed-assertion",
       FlutterErrorHelper.getAnalyticsId(
-        "'package:flutter/src/widgets/framework.dart':\nfailed assertion: line 123 pos 123: 'owner._debugcurrentbuildtarget == this': is not true"));
+        "'package:flutter/src/widgets/framework.dart':\n" +
+        "failed assertion: line 123 pos 123: 'owner._debugcurrentbuildtarget == this': is not true"));
     assertEquals(
       "unable-to-load-asset",
       FlutterErrorHelper.getAnalyticsId("unable to load asset: images/foobar.png"));
     assertEquals(
       "all-children-of-this-widget-must-have-a-key",
       FlutterErrorHelper.getAnalyticsId(
-        "all children of this widget must have a key.\n'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos xxx: 'children.every((widget w) => w.key != null)'"));
+        "all children of this widget must have a key.\n" +
+        "'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos xxx: 'children.every((widget w) => w.key != null)'"));
     assertEquals(
       "all-children-of-this-widget-must-have-a-key",
       FlutterErrorHelper.getAnalyticsId(
-        "all children of this widget must have a key. 'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos 123: 'children.every((widget w) => w.key != null)'"));
+        "all children of this widget must have a key.\n" +
+        "'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos 123: 'children.every((widget w) => w.key != null)'"));
     assertEquals(
       "could-not-find-a-generator-for-route-routesettings-in-the-_widgetsappstate",
       FlutterErrorHelper.getAnalyticsId("could not find a generator for route routesettings(\"/foobar\", null) in the _widgetsappstate"));
     assertEquals(
-      "controller's-length-property-does-not-match-the",
-      FlutterErrorHelper.getAnalyticsId("controller's length property (123) does not match the\nnumber of tabs"));
+      "controller's-length-property-does-not-match-the-number-of-tabs",
+      FlutterErrorHelper.getAnalyticsId("controller's length property (123) does not match the number of tabs"));
+  }
+
+  @Test
+  public void testGetAnalyticsId_errorStudiesApp() {
```

Comment:
@InMatrix - can you verify that the IDs here are roughly what we'd want to see for the given errors?

Reply:
Yes, those IDs look good. I'm not sure if we need to include the package information though. Is it included to disambiguate similar assertions?

Reply:
By the package info, do you mean the `package:flutter_error_studies/...` text in the source errors below? It's just the text that I copied in - it's nice to have real world examples (definitely an issue when I originally wrote this, to have to guess what the error might look like).

Reply:
No, I was referring to "'package:flutter/src/material/reorderable_list.dart': failed assertion: line 123 pos 123: 'children.every((widget w) => w.key != null)'"));" under "all children of this widget must have a key.\n" at line 56. I'm not sure if the extra detail about the specific file in package flutter and assertion position in the file are necessary for analytics .